core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -371,6 +442,12 @@ public Object doUnmarshal(final Object result, final HierarchicalStreamReader re
             OldDataMonitor.report((Saveable)result, (ArrayList<Throwable>)context.get("ReadError"));
             context.put("ReadError", null);
         }
+        if (existingObject) {
+            for (Map.Entry<FieldExpectation, Object> entry : expectedFields.entrySet()) {
+                reflectionProvider.writeField(result, entry.getKey().getName(), entry.getValue(),
```

Comment:
No, these are the fields that were not overwritten, and this is only for an already existing object.

I think the issue also affects non root objects only people have got away without noticing because of the root object blocker (or else the child objects are not updated)... need tests to verify further